cli/spcli/sectors.go

```diff
@@ -0,0 +1,1395 @@
+		numbers = append(numbers, uint64(sector))
+	}
+
+	return bitfield.NewFromSet(numbers)
+}
+
+func getSectorsFromFile(filePath string) ([]abi.SectorNumber, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	scanner := bufio.NewScanner(file)
+	sectors := make([]abi.SectorNumber, 0)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		id, err := strconv.ParseUint(line, 10, 64)
+		if err != nil {
+			return nil, xerrors.Errorf("could not parse %s as sector id: %s", line, err)
+		}
+
+		sectors = append(sectors, abi.SectorNumber(id))
+	}
+
+	if err = file.Close(); err != nil {
+		return nil, err
+	}
+
+	return sectors, nil
+}
+
+func NewPseudoExtendParams(p *miner.ExtendSectorExpiration2Params) (*PseudoExtendSectorExpirationParams, error) {
+	res := PseudoExtendSectorExpirationParams{}
+	for _, ext := range p.Extensions {
+		scount, err := ext.Sectors.Count()
+		if err != nil {
+			return nil, err
+		}
+
+		sectors, err := ext.Sectors.All(scount)
+		if err != nil {
+			return nil, err
+		}
+
+		res.Extensions = append(res.Extensions, PseudoExpirationExtension{
+			Deadline:      ext.Deadline,
+			Partition:     ext.Partition,
+			Sectors:       ArrayToString(sectors),
+			NewExpiration: ext.NewExpiration,
+		})
+	}
+	return &res, nil
+}
+
+type PseudoExtendSectorExpirationParams struct {
+	Extensions []PseudoExpirationExtension
+}
+
+type PseudoExpirationExtension struct {
+	Deadline      uint64
+	Partition     uint64
+	Sectors       string
+	NewExpiration abi.ChainEpoch
+}
+
+// ArrayToString Example: {1,3,4,5,8,9} -> "1,3-5,8-9"
+func ArrayToString(array []uint64) string {
+	sort.Slice(array, func(i, j int) bool {
+		return array[i] < array[j]
+	})
+
+	var sarray []string
+	s := ""
+
+	for i, elm := range array {
+		if i == 0 {
+			s = strconv.FormatUint(elm, 10)
+			continue
+		}
+		if elm == array[i-1] {
+			continue // filter out duplicates
+		} else if elm == array[i-1]+1 {
+			s = strings.Split(s, "-")[0] + "-" + strconv.FormatUint(elm, 10)
+		} else {
+			sarray = append(sarray, s)
+			s = strconv.FormatUint(elm, 10)
+		}
+	}
+
+	if s != "" {
+		sarray = append(sarray, s)
+	}
+
+	return strings.Join(sarray, ",")
+}
+
+func SectorsCompactPartitionsCmd(getActorAddress ActorAddressGetter) *cli.Command {
+	return &cli.Command{
+		Name:  "compact-partitions",
+		Usage: "removes dead sectors from partitions and reduces the number of partitions used if possible",
+		Flags: []cli.Flag{
+			&cli.Uint64Flag{
+				Name:     "deadline",
+				Usage:    "the deadline to compact the partitions in",
+				Required: true,
+			},
+			&cli.Int64SliceFlag{
+				Name:     "partitions",
+				Usage:    "list of partitions to compact sectors in",
+				Required: true,
+			},
+			&cli.BoolFlag{
+				Name:  "really-do-it",
+				Usage: "Actually send transaction performing the action",
+				Value: false,
+			},
+			&cli.StringFlag{
+				Name:  "actor",
+				Usage: "Specify the address of the miner to run this command",
+			},
```

Comment:
The main command flag supersedes this one
```suggestion
```

Reply:
Removed. Actor is top-level in both commands already.